### ISSUE TITLE
Accessing `stage.outs` required re-instantiatingthe class

### ISSUE THREAD

**Issue by PythonFZ** [zincware/py-track#39]
The following does not work:

```py
stage = StageIO()
stage(deps.resolve())
project.run()
project.load()

stage.outs.read_text()
```

while this does:

```py
stage = StageIO()
stage(deps.resolve())
project.run()
project.load()

stage = StageIO(id_=0) # <-- required!
stage.outs.read_text()
```

**Comment by PythonFZ** [comment]
Consider using a python descriptor to handle all dvc params following https://nbviewer.jupyter.org/urls/gist.github.com/ChrisBeaumont/5758381/raw/descriptor_writeup.ipynb

This way we can overwrite the `__set__` and `__get__` to load from the files

Dummy code:

```py

from weakref import WeakKeyDictionary

class NonNegative(object):
    """A descriptor that forbids negative values"""
    def __init__(self, default):
        self.default = default
        self.data = WeakKeyDictionary()

    def __get__(self, instance, owner):
        # we get here when someone calls x.d, and d is a NonNegative instance
        # instance = x
        # owner = type(x)
        return self.data.get(instance, self.default)

    def __set__(self, instance, value):
        # we get here when someone calls x.d = val, and d is a NonNegative instance
        # instance = x
        # value = val
        if value < 0:
            raise ValueError("Negative value not allowed: %s" % value)
        self.data[instance] = value
#%%


class Test:
    def __init__(self):
        self._value = None
        setattr(type(self), "value", NonNegative(0))

```

**Comment by PythonFZ** [comment]
fixed in #31
